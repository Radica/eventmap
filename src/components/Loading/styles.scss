.Loading {
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
    padding: 0 15px;
    height: 100%;
}

@keyframes animate-stripes {
    0% {
        background-position: 0 0;
    }

    100% {
        background-position: 60px 0;
    }
}

@keyframes auto-progress {
  0% {
    width: 0%;
  }

  100% {
    width: 100%;
  }
}

.ProgressBar {
    background-color: var(--darkgray);
    height: 20px;
    width: 50%;
    border-radius: 5px;
    box-shadow: 0 1px 5px #000 inset, 0 1px 0 #444;
}

.Stripes {
    --stripe-color: rgba(255, 255, 255, .15);
    background-size: 30px 30px;
    background-image: linear-gradient(
        135deg,
        var(--stripe-color) 25%,
        transparent 25%,
        transparent 50%,
        var(--stripe-color) 50%,
        var(--stripe-color) 75%,
        transparent 75%,
        transparent
    );
    animation: animate-stripes 0.6s linear infinite;
    animation-duration: 1.25s;
    animation-direction: reverse;
}
