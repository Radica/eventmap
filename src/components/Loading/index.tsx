import React from 'react';

import styles from './styles.scss';

export default () => (
    <div className={styles.Loading}>
        <p>Loading...</p>
    </div>
);
