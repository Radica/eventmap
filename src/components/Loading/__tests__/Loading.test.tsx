import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import Loading from '../index';

describe('<Loading />', () => {
    it('renders', () => {
        const tree = render(
            <MemoryRouter>
                <Loading />
            </MemoryRouter>
        ).container.firstChild;

        expect(tree).toMatchSnapshot();
    });
});
