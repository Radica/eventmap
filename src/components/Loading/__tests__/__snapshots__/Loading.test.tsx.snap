// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders 1`] = `
<div
  class="Loading"
>
  <p>
    Loading...
  </p>
</div>
`;
